### ISSUE TITLE
Hyperparameter Tuning with Ray Tune

### ISSUE THREAD

**Issue by mitunkantipaul** [mitunkantipaul/digit_recognizer#5]
- [ ] Set up Ray Tune for hyperparameter optimization.

- [ ] Define search space (learning rate, batch size, optimizer).

- [ ] Integrate with MLflow for logging.
